components/_util/wave.tsx

```diff
@@ -44,8 +47,10 @@ export default class Wave extends React.Component<{ insertExtraNode?: boolean }>
 
   context: ConfigConsumerProps;
 
+  private nodeReference = React.createRef<HTMLElement>();
+
   componentDidMount() {
-    const node = findDOMNode(this) as HTMLElement;
+    const node = this.nodeReference.current as HTMLElement;
```

Comment:
This is not equal to `findDOMNode`. Wave children is not always dom node.

Reply:
but it does the same effect, what is the benefits from findDOMNode, that we can't get after assigning the reference explicitly?

Reply:
@zombieJ also the tests are passing, which means it wont harm any children that doesn't include dom nodes.
I would like to solve the problem completely, can you guid me with more details?

Reply:
> but it does the same effect, what is the benefits from findDOMNode, that we can't get after assigning the reference explicitly?

`findDOMNode` always return dom node but `ref` is not.  It is not same effect.

You can consider to add prop in ConfigProvider to disable wave instead.